File: swarmmanagercontroller/src/main/java/co/uk/swarmbit/docker/api/tasks/parameters/TasksListParameters.java
package co.uk.swarmbit.docker.api.tasks.parameters;

import co.uk.swarmbit.util.EncoderDecoder;
import co.uk.swarmbit.docker.api.common.parameters.FiltersParameters;
import co.uk.swarmbit.rest.QueryParam;

import java.util.Optional;

public class TasksListParameters implements FiltersParameters {

    private static final String FILTERS_NAME = "filters";

    private Optional<QueryParam> filters;

    public TasksListParameters() {
        filters = Optional.empty();
    }

    public Optional<QueryParam> getFilters() {
        return filters;
    }

    public void setFilters(TasksFilters filters) {
        this.filters = Optional.of(new QueryParam(FILTERS_NAME, EncoderDecoder.jsonUrlEncode(filters.getJson())));
    }
}

File: swarmmanagercontroller-new/src/main/kotlin/com/swarmbit/docker/api/tasks/parameters/TasksListParameters.kt
package com.swarmbit.docker.api.tasks.parameters

import com.swarmbit.docker.api.common.parameters.FiltersParameters
import com.swarmbit.docker.api.common.rest.model.QueryParam

class TasksListParameters : FiltersParameters {

    companion object {
        private const val FILTERS_NAME = "filters"
    }

    private var innerFilters: QueryParam? = null

    fun setFilters(filters: TasksFilters): TasksListParameters {
        this.innerFilters = QueryParam(FILTERS_NAME, filters.asUrlJson())
        return this
    }

    override val filters: QueryParam?
        get() = innerFilters
}
